rct229/rulesets/ashrae9012019/ruleset_functions/get_lab_zone_hvac_systems_test.py

```diff
@@ -0,0 +1,554 @@
+                                            "served_by_heating_ventilating_air_conditioning_system": "Testing Sys",
+                                        }
+                                    ],
+                                },
+                            ],
+                            "heating_ventilating_air_conditioning_systems": [
+                                {
+                                    "id": "Testing Sys",
+                                    "heating_system": {
+                                        "id": "Preheat Coil 1",
+                                        "type": "FURNACE",
+                                    },
+                                    "cooling_system": {
+                                        "id": "CHW Coil 1",
+                                        "type": "NONE",
+                                    },
+                                }
+                            ],
+                        }
+                    ],
+                }
+            ],
+        }
+    ],
+}
+
+TEST_RMD_B_FULL_G311D_ONE_SYS = {
+    "id": "229",
+    "ruleset_model_descriptions": [TEST_RMD_B_ALL_ZONES_G311D_ONE_SYSTEM],
+}
+TEST_RMD_B_UNIT_G311D_ONE_SYS = quantify_rmr(TEST_RMD_B_FULL_G311D_ONE_SYS)[
+    "ruleset_model_descriptions"
+][0]
+TEST_RMD_B_FULL_G311D_TWO_SYS = {
+    "id": "229",
+    "ruleset_model_descriptions": [TEST_RMD_B_ALL_ZONES_G311D_ONE_SYSTEM2_TWO_SYSTEMS],
+}
+TEST_RMD_B_UNIT_G311D_TWO_SYS = quantify_rmr(TEST_RMD_B_FULL_G311D_TWO_SYS)[
+    "ruleset_model_descriptions"
+][0]
+
+TEST_RMD_B_FULL_G311D_G311F = {
+    "id": "229",
+    "ruleset_model_descriptions": [TEST_RMD_B_G311D_G311F],
+}
+TEST_RMD_B_UNIT_G311D_G311F = quantify_rmr(TEST_RMD_B_FULL_G311D_G311F)[
+    "ruleset_model_descriptions"
+][0]
+
+TEST_RMD_UNIT_P = quantify_rmr(TEST_RMD_P)["ruleset_model_descriptions"][0]
+
+
+def test__TEST_RMD_B_ALL_ZONES_G311D_ONE_SYSTEM__is_valid():
+    schema_validation_result = schema_validate_rmr(TEST_RMD_B_FULL_G311D_ONE_SYS)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__TEST_RMD_B_ALL_ZONES_G311D_ONE_SYSTEM2_TWO_SYSTEMS__is_valid():
+    schema_validation_result = schema_validate_rmr(TEST_RMD_B_FULL_G311D_TWO_SYS)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__TEST_RMD_P_is_valid():
+    schema_validation_result = schema_validate_rmr(TEST_RMD_P)
+    assert schema_validation_result[
+        "passed"
+    ], f"Schema error: {schema_validation_result['error']}"
+
+
+def test__get_lab_zone_hvac_systems_all_zones_g311d_one_system__true():
+    assert get_lab_zone_hvac_systems(
+        TEST_RMD_B_UNIT_G311D_ONE_SYS, TEST_RMD_UNIT_P, "CZ4A", False
+    ) == {
+        "lab_zones_only": ["Thermal Zone 1", "Thermal Zone 2"],
+        "lab_and_other": [],
+    }
```

Comment:
You will have to update this dictionary because it saves HVAC_ID, not thermal zones.

Reply:
Addressed.